bin/propolis-standalone/src/main.rs

```diff
@@ -404,8 +406,17 @@ impl Instance {
                     for mut vcpu_ctrl in guard.vcpu_tasks.drain(..) {
                         vcpu_ctrl.exit();
                     }
+                    if guard.exit_code.is_none() {
+                        guard.exit_code = Some(inner.config.main.exit_on_halt);
+                    }
                 }
                 State::Reset => {
+                    if let Some(code) = inner.config.main.exit_on_reboot {
+                        guard.exit_code = Some(code);
```

Comment:
Should we do the same **is_none()** check you're doing on 409 for the **Halt** version?

Reply:
I don't believe it's strictly necessary at this time, but for defense-in-depth purposes, I've added such a check. (In case we add something like exit-on-fault to bail out on a triple-fault or something)